File: Frontend/src/types/Product.ts


/**
 * 상품 타입
 * @name 상품 이름
 * @description 상품 설명
 * @price 상품 가격
 * @shippingPrice 배송비
 * @shippingDay 배송일
 * @inventory 재고
 */
interface Product {
    id?: number;
    name: string;
    description: string;
    price: number;
    shippingPrice: number;
    shippingDay: string;
    inventory: number;
}

export default Product;

File: Backend/src/main/java/com/example/demo/Controller/ProjectController/DTO/Product.java
package com.example.demo.Controller.ProjectController.DTO;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Product {
    private Long id;
    //private String preview;
    private String name;
    private String description;
    private Integer price;
    private Integer shippingPrice;
    private String shippingDay;
    private Integer inventory;
}
